extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -113,4 +110,28 @@ public void close()
     unions.clear();
   }
 
+  @Override
+  public void relocate(int oldPosition, int newPosition, ByteBuffer newBuffer)
```

Comment:
Wouldn't `relocate(int[] newPositions, ByteBuffer newBuffer)` make what @cheddar wanted to do with quantiles (see https://github.com/druid-io/druid/issues/4026#issuecomment-285202370) impossible?

In that proposal, datasketches quantiles aggregator wouldn't just be caching things that could be recomputed, it would be storing state that needs to be preserved.